### PR TITLE
waterfall transform errors

### LINKED ISSUE #3719: Error with transforms in waterfall trace
https://codepen.io/dmt0/pen/NmGGRa
See console for errors.

### PR DESCRIPTION
**Pull request by archmoj**
Fix #3719 
@etpinard

### CODE REVIEW

File: src/traces/bar/plot.js [hunk 1]
```diff
@@ -107,10 +107,12 @@ module.exports = function plot(gd, plotinfo, cdModule, traceLayer) {
...
             if(trace.type === 'waterfall') {
-                var cont = trace[di.dir].marker;
-                lw = cont.line.width;
-                mc = cont.color;
-                prefix = 'waterfall';
+                if(!isBlank) {
+                    var cont = trace[di.dir].marker;
+                    lw = cont.line.width;
+                    mc = cont.color;
+                    prefix = 'waterfall';
```
Comment (etpinard) [inline src/traces/bar/plot.js]:
```
 prefix = 'waterfall';
```

is still well defined when `isBlank:true`, I'd leave it out of this block.

File: test/jasmine/tests/waterfall_test.js [hunk 2]
```diff
@@ -948,6 +948,36 @@ describe('A waterfall plot', function() {
         .then(done);
     });
 
+    it('should be able to deal with blank bars on transform', function(done) {
```
Comment (etpinard) [inline test/jasmine/tests/waterfall_test.js]: 10/10 test. Nice!

**Comment by etpinard** [comment]
:dancer: https://github.com/plotly/plotly.js/pull/3720#discussion_r271990937 is non-blocking.

File: src/traces/bar/plot.js (hunk 1 above)
Comment (archmoj) [reply src/traces/bar/plot.js]: Thanks for spotting that! Done in https://github.com/plotly/plotly.js/pull/3720/commits/be51cafe2d1d825ea1f3a3240a9f59c6c5cfa5ed.

**Pull request merged**